sashimi/processes/dispatcher.py

```diff
@@ -31,37 +34,38 @@ def __init__(
         self.calibration_ref_queue = ArrayQueue()
 
         self.volume_buffer = None
-        self.current_frame = None
         self.calibration_ref = None
 
         self.n_planes = 1
         self.i_plane = 0
         self.first_volume = True
 
+        self.k = 0
+
     def run(self):
         self.logger.log_message("started")
         while not self.stop_event.is_set():
             self.send_receive()
             self.get_frame()
         self.close_log()
 
-    def process_frame(self):
-        if self.current_frame is not None:
+    def process_frame(self, current_frame=None):
+        if current_frame is not None:
```

Comment:
"flat is better than nested": invert the condition and just return immediately if it is None